Task_2_1_1/src/main/java/r/masud/parallel.java

```diff
@@ -0,0 +1,70 @@
+package r.masud;
+
+public class parallel {
+    // non - prime number in parallel solution using the Thread class
+    class NonPrimeChecker {
+        public boolean containsNonPrime(int[] nums, int numThreads) {
+            Thread[] threads = new Thread[numThreads];
+            int chunkSize = (int) Math.ceil((double) nums.length / numThreads);
+
+            NonPrimeThread[] nonPrimeThreads = new NonPrimeThread[numThreads];
+            for (int i = 0; i < numThreads; i++) {
+                int start = i * chunkSize;
+                int end = Math.min(start + chunkSize, nums.length);
+                nonPrimeThreads[i] = new NonPrimeThread(nums, start, end);
+                threads[i] = new Thread(nonPrimeThreads[i]);
+                threads[i].start();
+            }
+
+            for (int i = 0; i < numThreads; i++) {
+                try {
+                    threads[i].join();
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+
+            for (NonPrimeThread nonPrimeThread : nonPrimeThreads) {
```

Comment:
Implement early-exit mechanism in case if any of threads found non-prime number way too faster than others

It can be achieved with shared AtomicBoolean flag of local thread flags read by main thread and interruption of non-finished threads